app/routes/task_routes.py

```diff
@@ -1 +1,118 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, Response, abort, make_response
+from ..models.task import Task
+from ..db import db
+from datetime import datetime
+import requests
+import os
+
+
+bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+# make a POST request to /tasks with the following HTTP request body
+@bp.post("")
+def create_task():
+    request_body = request.get_json()
+    new_task = Task.obj_from_dict(request_body)
+    
+    db.session.add(new_task)
+    db.session.commit()
+ 
+    response = {"task": new_task.obj_to_dict()}
+
+    return response, 201
+
+# GET request to /tasks
+@bp.get("")
+def get_all_task():
+    query = db.select(Task)
+    sort =  request.args.get("sort")
+
+    if sort == "asc":
+        query = query.order_by(Task.title.asc())
+    elif sort == "desc":
+        query = query.order_by(Task.title.desc())
+    else:
+        query = query.order_by(Task.id)
+    tasks = db.session.scalars(query)
+
+    tasks_response = [task.obj_to_dict() for task in tasks]
+
+    return tasks_response
+
+@bp.get("/<task_id>")
+def get_one_task(task_id):
+    task = validate_model(Task, task_id)
+
+    return {"task": task.obj_to_dict()}, 200
+
+@bp.put("/<task_id>")
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    
+    db.session.commit()
+
+    return { "task": task.obj_to_dict()}
+
+@bp.patch("/<task_id>/mark_complete")
+def update_complete(task_id):
+    task = validate_model(Task, task_id)
+    task.completed_at = datetime.utcnow()
+
+    db.session.commit()
+
+    path = "https://slack.com/api/chat.postMessage"
+    token = os.environ.get("SLACK_BOT_TOKEN")
+    channel_id = "C07V4J7ABF1"
```

Comment:
Nice job using variables to reference these values so you don't have string literals embedded below. Since these are constant variables, they should be named with all capital letters.